speech/speech_samples.rb

```diff
@@ -12,160 +12,168 @@
 
-  speech = Google::Cloud::Speech.new project: project_id
-  audio  = speech.audio audio_file_path, encoding:    :linear16,
-                                         sample_rate: 16000,
-                                         language:    "en-US"
+  speech = Google::Cloud::Speech.new
 
-  results = audio.recognize words: true
+  audio_file = File.binread audio_file_path
+  config     = { encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US", enable_word_time_offsets: true }
+  audio      = { content: audio_file }
 
-  results.each do |result|
-    puts "Transcription: #{result.transcript}"
+  response = speech.recognize config, audio
+
+  alternatives = response.results.first.alternatives
 
-    result.words.each do |word|
-      puts "Word: #{word.word} #{word.start_time} #{word.end_time}"
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
+
+    alternative.words.each do |word|
+      start_time = word.start_time.seconds + word.start_time.nanos/1000000000.0
+      end_time   = word.end_time.seconds + word.end_time.nanos/1000000000.0
+
+      puts "Word: #{word.word} #{start_time} #{end_time}"
     end
   end
 # [END speech_sync_recognize_words]
 end
 
-def speech_sync_recognize_gcs project_id: nil, storage_path: nil
+def speech_sync_recognize_gcs storage_path: nil
 # [START speech_sync_recognize_gcs]
-  # project_id   = "Your Google Cloud project ID"
   # storage_path = "Path to file in Cloud Storage, eg. gs://bucket/audio.raw"
 
   require "google/cloud/speech"
 
-  speech = Google::Cloud::Speech.new project: project_id
-  audio  = speech.audio storage_path, encoding:    :linear16,
-                                      sample_rate: 16000,
-                                      language:    "en-US"
+  speech = Google::Cloud::Speech.new
 
-  results = audio.recognize
+  config = { encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US" }
+  audio  = { uri: storage_path }
 
-  results.each do |result|
-    puts "Transcription: #{result.transcript}"
+  response = speech.recognize config, audio
+
+  alternatives = response.results.first.alternatives
+
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
   end
 # [END speech_sync_recognize_gcs]
 end
 
-def speech_async_recognize project_id: nil, audio_file_path: nil
+def speech_async_recognize audio_file_path: nil
 # [START speech_async_recognize]
-  # project_id      = "Your Google Cloud project ID"
   # audio_file_path = "Path to file on which to perform speech recognition"
 
   require "google/cloud/speech"
 
-  speech = Google::Cloud::Speech.new project: project_id
-  audio  = speech.audio audio_file_path, encoding:    :linear16,
-                                         sample_rate: 16000,
-                                         language:    "en-US"
+  speech = Google::Cloud::Speech.new
 
-  operation = audio.process
+  audio_file = File.binread audio_file_path
```

Comment:
It would of course be dope if we could accept a local file path 😄 

For example y'all ...

```ruby
def recognize **kwargs
  audio = {}
  if kwargs[:audio_uri]
    audio[:uri] = kwargs.delete :audio_uri
  elsif kwargs[:audio_content]
    audio[:content] = kwargs.delete :audio_content
  elsif kwargs[:audio_path]
    audio[:content] = File.binread kwargs.delete(:audio_path)
  else
    raise "No dawg, u needs audio yo! 🐕"
  end

  super audio, kwargs
end
```

Reply:
Hey hey @frankyn et al ... Look! An idea for Path! How easy, amiright??

@jgeiger @lukesneeringer

Reply:
# 😇